superset/assets/javascripts/explore/explore.jsx

```diff
@@ -73,6 +73,7 @@ function query(forceUpdate, pushState) {
     // update the url after prepForm() fix the field ids
     history.pushState({}, document.title, slice.querystring());
   }
+  slice.container.html('');
```

Comment:
I added this line because I noticed some strange behavior in box-plot and pie charts when I changed major parts of the query such as the the dimension I was grouping by.

Example of strange behavior:
- Pie chart not resizing properly when hovered.

On a dashboard, the same query would always be firing from each slice so it shouldn't be an issue.